### PR TITLE
fix: title generation fails with OpenAI reasoning models

### LINKED ISSUE #2318: Title generation with gpt-5 fails
```
docker agent debug title ./examples/rag.yaml "What can you do?"
```

gives use this error:

```
generating title: generating title failed: empty title output from model "openai/gpt-5"
```

### PR DESCRIPTION
**Pull request by trungutt**
- Title generation with OpenAI reasoning models (gpt-5, o-series) fails with `empty title output from model` because these models always reason internally, even when no reasoning configuration is provided. The hidden reasoning tokens consume the entire output token budget, leaving nothing for the visible title text.

- When the code intended to "skip reasoning", it simply omitted the reasoning parameter from the API request. This does not disable reasoning - the model defaults to its standard reasoning behavior, which can easily exhaust a small token budget.

- The output token budget (20 tokens) was sized for non-reasoning models and far too small once reasoning overhead is factored in.

## Fix

- Explicitly request the lowest reasoning effort supported across all OpenAI reasoning models so the model minimizes internal reasoning token usage.
- Increase the output token budget to accommodate the reasoning overhead that reasoning models always incur.

## How to test

```bash
mise build

# The original failing command from the issue
./bin/docker-agent debug title ./examples/rag.yaml "What can you do?"

# Explicitly test reasoning models
./bin/docker-agent debug title ./examples/rag.yaml --model=openai/gpt-5 "What can you do?"
./bin/docker-agent debug title ./examples/rag.yaml --model=openai/o3-mini "What can you do?"
./bin/docker-agent debug title ./examples/rag.yaml --model=openai/o4-mini "What can you do?"

# Non-reasoning models should still work as before
./bin/docker-agent debug title ./examples/rag.yaml --model=openai/gpt-4o "What can you do?"
./bin/docker-agent debug title ./examples/rag.yaml --model=anthropic/claude-haiku-4-5 "What can you do?"
```

Each command should print a short title to stdout without errors.

Fixes #2318

### CODE REVIEW

**Comment by dgageot** [comment]
@trungutt can you add an e2e test that makes sure it generates a non empty title?

File: pkg/sessiontitle/generator.go [hunk 1]
```diff
@@ -103,7 +109,7 @@ func (g *Generator) Generate(ctx context.Context, sessionID string, userMessages
 			ctx,
 			baseModel,
 			options.WithStructuredOutput(nil),
-			options.WithMaxTokens(20),
+			options.WithMaxTokens(titleMaxTokens),
```
Comment (dgageot) [inline pkg/sessiontitle/generator.go:112]: we go from 20 to 256. That's probably not needed for every model

**Review by aheritier — APPROVED** [review]
